linker/shared/src/main/scala/org/scalajs/linker/frontend/optimizer/OptimizerCore.scala

```diff
@@ -1762,7 +1765,7 @@ private[optimizer] abstract class OptimizerCore(config: CommonPhaseConfig) {
               else argsNoSpread ::: List.fill(missingArgCount)(Undefined())
             pretransformExprs(expandedArgs) { targs =>
               inlineBody(
-                  Some(PreTransLit(Undefined())), // `this` is `undefined`
+                  Some((AnyType, PreTransLit(Undefined()))), // `this` is `undefined`
```

Comment:
Hum, because the declared type of the `this` of a JS function is `any`, like all the regular parameters. It would be inconsistent to use anything else here.